tensorflow/compiler/mlir/xla/transforms/map_lhlo_to_scalar_op.h

```diff
@@ -148,44 +155,173 @@ inline Optional<CmpIPredicate> getIntCmpPredicate(
     ArrayRef<Value*> block_args, OpBuilder b) {
-  return b.create<::mlir::SelectOp>(lhlo_op.getLoc(), result_types, block_args,
-                                    mlir::None);
+  Operation* op = b.create<::mlir::SelectOp>(lhlo_op.getLoc(), result_types,
+                                             block_args, mlir::None);
+  return op->getResult(0);
+}
+
+template <>
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::AbsOp>(
+    xla_lhlo::AbsOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  Type element_type = block_args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    Operation* op = b.create<::mlir::AbsFOp>(lhlo_op.getLoc(), result_types,
+                                             block_args, mlir::None);
+    return op->getResult(0);
+  }
+  return nullptr;
 }
 
 template <>
-inline Operation* MapLhloOpToStdScalarOp<xla_lhlo::ExpOp>(
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::CeilOp>(
+    xla_lhlo::CeilOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  Type element_type = block_args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    Operation* op = b.create<::mlir::CeilFOp>(lhlo_op.getLoc(), result_types,
+                                              block_args, mlir::None);
+    return op->getResult(0);
+  }
+  return nullptr;
+}
+
+template <>
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::CosOp>(
+    xla_lhlo::CosOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  Type element_type = block_args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    Operation* op = b.create<::mlir::CosOp>(lhlo_op.getLoc(), result_types,
+                                            block_args, mlir::None);
+    return op->getResult(0);
+  }
+  return nullptr;
+}
+
+template <>
+inline Value* MapLhloOpToStdScalarOp<xla_lhlo::ConvertOp>(
+    xla_lhlo::ConvertOp lhlo_op, ArrayRef<Type> result_types,
+    ArrayRef<Value*> block_args, OpBuilder b) {
+  const Type& sourceType = block_args.front()->getType();
+  const Type& targetType = result_types.front();
+
+  Operation* op;
+  if (mlir::SIToFPOp::areCastCompatible(sourceType, targetType)) {
+    op = b.create<mlir::SIToFPOp>(lhlo_op.getLoc(), result_types, block_args,
+                                  mlir::None);
+  } else if (sourceType.isa<FloatType>() && targetType.isa<FloatType>()) {
+    FloatType src = sourceType.cast<FloatType>();
+    FloatType res = targetType.cast<FloatType>();
+    if (src.getWidth() > res.getWidth()) {
+      op = b.create<mlir::FPTruncOp>(lhlo_op.getLoc(), result_types, block_args,
+                                     mlir::None);
+    } else if (src.getWidth() < res.getWidth()) {
+      op = b.create<mlir::FPExtOp>(lhlo_op.getLoc(), result_types, block_args,
+                                   mlir::None);
+    } else {
+      return block_args.front();
```

Comment:
`block_args` is a really confusing name here. I know this is used in the entire file, probably for historic reasons. `args` would be much better.